addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -0,0 +1,314 @@
+                        'name': record.l10n_id_tax_number,
+                        'invoice_id': [(6, 0, record.ids)],
+                        'company_id': record.company_id.id,
+                    })
+            elif record.l10n_id_efaktur_id.name != record.l10n_id_tax_number:
+                raise UserError(_("You can't change the number manually for a Customer Invoice"))
+
+    @api.depends('l10n_id_efaktur_id')
+    def _compute_tax_number(self):
+        for record in self:
+            record.l10n_id_tax_number = record.l10n_id_efaktur_id.name
+
+    @api.depends('l10n_id_attachment_id')
+    def _compute_csv_created(self):
+        for record in self:
+            record.l10n_id_csv_created = bool(record.l10n_id_attachment_id)
+
+    @api.depends('partner_id', 'line_ids.tag_ids')
+    def _compute_need_kode_transaksi(self):
+        for move in self:
+            move.l10n_id_need_kode_transaksi = move.partner_id.l10n_id_pkp and not move.l10n_id_efaktur_id and move.type == 'out_invoice'
+
+    @api.constrains('l10n_id_kode_transaksi', 'line_ids')
+    def _constraint_kode_ppn(self):
+        ppn_tag = self.env.ref('l10n_id.ppn_tag')
+        for move in self.filtered(lambda m: m.l10n_id_kode_transaksi != '08'):
+            if any(ppn_tag.id in line.tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False) and any(ppn_tag.id not in line.tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False):
+                raise UserError(_('Cannot mix VAT subject and Non-VAT subject items in the same invoice with this kode transaksi.'))
+        for move in self.filtered(lambda m: m.l10n_id_kode_transaksi == '08'):
+            if any(ppn_tag.id in line.tag_ids.ids for line in move.line_ids if line.exclude_from_invoice_tab is False):
+                raise UserError('Kode transaksi 08 is only for non VAT subject items.')
+
+    def post(self):
+        """Set E-Faktur number after validation."""
+        for move in self:
+            if move.l10n_id_need_kode_transaksi:
+                if not move.l10n_id_kode_transaksi:
+                    raise ValidationError(_('You need to put a Kode Transaksi for this partner.'))
+                if move.l10n_id_replace_invoice_id.l10n_id_efaktur_id:
+                    if not move.l10n_id_replace_invoice_id.l10n_id_date_validated:
+                        raise ValidationError(_('Replacement invoice only for invoices on which e-faktur is validated'))
+                    rep_efaktur_str = move.l10n_id_replace_invoice_id.l10n_id_efaktur_id.name
+                    new_efaktur_str = '%s1%s' % (move.l10n_id_kode_transaksi, rep_efaktur_str[3:])
+                    efaktur = self.env['l10n_id_efaktur.efaktur'].search([('name', '=', new_efaktur_str)]) or self.env['l10n_id_efaktur.efaktur'].create({
+                        'name': new_efaktur_str,
+                        'invoice_id': [(6, 0, move.ids)],
```

Comment:
invoice_id or invoice_ids?

Reply:
`invoice_id,` there is only one move 
```python
        for move in self:
            ...
```

I used that syntax because I know it and I didn't want to look at the documentation for other ORM commands, and it is strictly equivalent.